clients/rust/src/lib.rs

```diff
@@ -397,88 +613,246 @@ impl ClientExt for Client {
         create_datapoints_internal(self, dataset_name, params, "datapoints/bulk").await
     }
 
+    /// DEPRECATED: Use `list_datapoints` instead.
```

Comment:
do we want to put the deprecation tag here too?

Reply:
#[deprecated] only works on the trait declarations; on the implementations they have no effect. (Clippy will complain.)